TELCODOCS-2036: Procedure added for MLX secure boot

ogradyp opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.
Do not create or rename a top-level directory (or any subdirectory in a directory that contains a hugebook.flag file) in the repository and topic map without checking with a docs program manager first.
If a book is being created or modified, there are changes on the Customer Portal that must also be made.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

cgoncalves commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,79 @@
+
+.. Reboot the system to enable the virtual functions.
+
+. Configure the sriov operator with the Mellanox plugin disabled
```
```suggestion
. Configure the SR-IOV Network Operator with the Mellanox plugin disabled
```

ogradyp replied:
Thanks @cgoncalves for your feedback. I have applied your suggestion.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,98 @@
+[id="nw-sriov-nic-mlx-secure-boot_{context}"]
+= MLX Secure Boot
+
+The Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.
```
Updated with the ocpdocs-vale-bot recommendation.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,98 @@
+[id="nw-sriov-nic-mlx-secure-boot_{context}"]
+= MLX Secure Boot
+
+The Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.
```
Updated with the ocpdocs-vale-bot recommendation.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,79 @@
+[id="nw-sriov-nic-mlx-secure-boot_{context}"]
+= Mellanox Secure Boot
+
+Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.
```
Updated with the ocpdocs-vale-bot recommendation.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,79 @@
+[id="nw-sriov-nic-mlx-secure-boot_{context}"]
+= Mellanox Secure Boot
+
+Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.
```
Updated with the ocpdocs-vale-bot recommendation.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,79 @@
+[id="nw-sriov-nic-mlx-secure-boot_{context}"]
+= Mellanox Secure Boot
+
+Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.
```
Thanks @cgoncalves for your feedback. I have applied your suggestions.

SchSeba commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,78 @@
+
+// :_mod-docs-content-type: PROCEDURE
+[id="nw-sriov-nic-mlx-secure-boot_{context}"]
+= Mellanox Secure Boot
```
maybe:  Configure SR-IOV on Mellanox cards when Secure Boot is enabled

ogradyp replied:
Thanks @SchSeba for your feedback. I have renamed the title 'Configuring the SR-IOV Network Operator on Mellanox cards when Secure Boot is enabled'

SchSeba commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,78 @@
+[id="nw-sriov-nic-mlx-secure-boot_{context}"]
+= Mellanox Secure Boot
+
+Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV Network Operator. This allows {product-title} customers to use the existing virtual functions in the Mellanox cards firmware, and is currently the only way to use Mellanox NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.
```
I think here it should be something like.

The SR-IOV operator supports an option to skip the firmware configuration for Mellanox devices. That is currently the only way to be able and create virtual functions via the SR-IOV Network operator if the system has secure boot enabled.

This will require a manual configuration from the user to allocate the number of virtual functions in the firmware before switching the system to secure boot.


Note: the number of virtual functions in the firmware will be the max number of virtual functions the user can request in the policy (meaning the user can create less)

ogradyp replied:
Thanks @SchSeba for your feedback. I have rewritten the opening paragraph and added the Note.

SchSeba commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,78 @@
+
+.. Reboot the system to enable the virtual functions.
+
+. Configure the SR-IOV Network Operator with the Mellanox plugin disabled.
```
maybe switching the section to configure secure boot after this change (and reboot the system) this will reduce the number if required reboots by one

ogradyp replied:
Thanks @SchSeba for your feedback. I have applied your suggestion.

SchSeba reviewed (commented):
nice work!

I left some small comments thanks! :)

SchSeba commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,83 @@
++
+[source,terminal]
+----
+mstconfig -d -0001:b1:00.0 set SRIOV_EN=1 NUM_OF_VFS=16
```
`-0001:b1:00.0` -> `0001:b1:00.0`

also can you put a small note about the parameters?

1. <PF-pci-address> 
2. enable sriov support on the card
3. number of virtual functions to enable in the card firmware

SchSeba commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,83 @@
++
+[source,terminal]
+----
+# oc -n openshift-sriov-network-operator get sriovnetworknodestate.sriovnetwork.openshift.io  cnfdr14.telco5g.eng.rdu2.redhat.com -oyaml
```
lets not expose our server names `cnfdr14.telco5g.eng.rdu2.redhat.com` -> `worker-0`

ogradyp replied:
Thanks @SchSeba for your feedback. I have applied your suggestion.

SchSeba commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,83 @@
+    mtu: 1500
+    name: ens3f1np1
+    pciAddress: 0000:b1:00.1
+    totalvfs: 16
```
please add a note to point this is the number that was configured in the mstconfig for the firmware

SchSeba replied:
@ogradyp not the `pciAddress` it's the `totalvfs` it should be the same number that the customer apply in `NUM_OF_VFS=16` 

command from above

`mstconfig -d -0001:b1:00.1 set SRIOV_EN=1 NUM_OF_VFS=16 <1> <2>`

ogradyp replied:
Thanks @SchSeba for your feedback. I have applied your suggestion.

SchSeba reviewed (commented):
just small comments in general looks good nice work! :)

SchSeba reviewed (approved):
/lgtm

ogradyp commented:
/label peer-review-needed

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
+//
+// * networking/hardware_networks/configuring-sriov-device.adoc
+
+// :_mod-docs-content-type: PROCEDURE
```
Thanks @xenolinux for your feedback. I have applied your suggestion.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
+
+[NOTE]
+====
+The number of virtual functions in the firmware will be the maximum number of virtual functions that you can request in the policy.
```
Thanks @xenolinux for your feedback. I have applied your suggestion.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
+
+.Procedure
+
+. Configure the virtual functions (VFs).
```
Thanks @xenolinux for your feedback. I have applied your suggestion.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
++
+[source,terminal]
+----
+mstconfig -d -0001:b1:00.1 set SRIOV_EN=1 NUM_OF_VFS=16 <1> <2>
```
Thanks @xenolinux for your feedback. I have applied your suggestion.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
+mstconfig -d -0001:b1:00.1 set SRIOV_EN=1 NUM_OF_VFS=16 <1> <2>
+----
+<1> `SRIOV_EN` Enable SR-IOV Network operator support on the Mellanox card.
+<2> `NUM_OF_VFS` Specify the number of virtual functions to enable in the firmware.
```
Thanks @xenolinux for your feedback. I have applied your suggestion.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
+  logLevel: 2
+----
+
+. Reboot the system to enable the virtual functions and the configuration settings.
```
Thanks @xenolinux for your feedback. Yes reboot is suitable here.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
+[id="nw-sriov-nic-mlx-secure-boot_{context}"]
+= Configuring the SR-IOV Network Operator on Mellanox cards when Secure Boot is enabled
+
+The SR-IOV Network Operator supports an option to skip the firmware configuration for Mellanox (MLX) devices. It is currently the only way to create virtual functions by using the SR-IOV Network operator, if the system has secure boot enabled. You are required to manually configure and allocate the number of virtual functions in the firmware before switching the system to secure boot.
```
Thanks @xenolinux for your feedback. All relevant instances of Operator and now capitalized.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
+<1> `SRIOV_EN` Enable SR-IOV Network operator support on the Mellanox card.
+<2> `NUM_OF_VFS` Specify the number of virtual functions to enable in the firmware.
+
+. Configure the SR-IOV Network Operator with the Mellanox plugin disabled.
```
Thanks @xenolinux for your feedback. I have applied your suggestion.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
++
+[source,terminal]
+----
+# oc -n openshift-sriov-network-operator get sriovnetworknodestate.sriovnetwork.openshift.io worker-0 -oyaml
```
Thanks @xenolinux for your feedback. I have applied your suggestion.

ogradyp commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,86 @@
+----
+<1> The `totalvfs` value is the same number used in the `mstconfig` command earlier in the procedure. 
+
+. Enable secure boot
+Enabling secure boot improves your system's security by preventing unauthorized operating systems and malicious software from loading during the device's boot process. 
+
+.. Enable secure boot using the BIOS.
++
+[source,yaml]
+----
+Secure Boot: Enabled
+Secure Boot Policy: Standard
+Secure Boot Mode: Mode Deployed
+----
+
+.. Reboot the machine.
```
Thanks @xenolinux for your feedback. I have expanded BIOS, added [source,terminal], and replaced machine with system.

ogradyp commented:
> Posted couple of comments and questions to consider.
> 
> General comments:
> 
>     * Gentle reminder to squash commits
> 
>     * Make sure to use the terms `machine` and `system` consistently in the PR.
> 
>     * Make sure to capitalize the "O" in the Operator throughout the PR.
> 
>     * You might want to check with SME regarding using `restart` vs `reboot`.

Thanks @xenolinux for your feedback. I have squashed commits and addressed all your comments..

xenolinux commented:
> > Posted couple of comments and questions to consider.
> > General comments:
> > ```
> > * Gentle reminder to squash commits
> > 
> > * Make sure to use the terms `machine` and `system` consistently in the PR.
> > 
> > * Make sure to capitalize the "O" in the Operator throughout the PR.
> > 
> > * You might want to check with SME regarding using `restart` vs `reboot`.
> > ```
> 
> Thanks @xenolinux for your feedback. I have squashed commits and addressed all your comments..

Thank you @ogradyp Appreciate it.

ogradyp commented:
/label merge-review-needed

jeana-redhat commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,85 @@
+[id="nw-sriov-nic-mlx-secure-boot_{context}"]
+= Configuring the SR-IOV Network Operator on Mellanox cards when Secure Boot is enabled
+
+The SR-IOV Network Operator supports an option to skip the firmware configuration for Mellanox (MLX) devices. It is currently the only way to create virtual functions by using the SR-IOV Network Operator, if the system has secure boot enabled. You are required to manually configure and allocate the number of virtual functions in the firmware before switching the system to secure boot.
```
I don't see "MLX" in any official docs or web presence for this brand
```suggestion
The SR-IOV Network Operator supports an option to skip the firmware configuration for Mellanox devices. This option allows you to create virtual functions by using the SR-IOV Network Operator when the system has secure boot enabled. You must manually configure and allocate the number of virtual functions in the firmware before switching the system to secure boot.
```

ogradyp replied:
Thanks @jeana-redhat for your feedback. I have applied your suggestion.

jeana-redhat commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,85 @@
+
+[NOTE]
+====
+The number of virtual functions in the firmware are the maximum number of virtual functions that you can request in the policy.
```
```suggestion
The number of virtual functions in the firmware is the maximum number of virtual functions that you can request in the policy.
```

ogradyp replied:
Thanks @jeana-redhat for your feedback. I have applied your suggestion.

jeana-redhat commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,85 @@
+
+.Procedure
+
+. Configure the virtual functions (VFs):
+
+.. Run the following command when the system is without a secure boot when using the sriov-config daemon:
```
this step has a single substep - this should just be a step.

ogradyp replied:
Thanks @jeana-redhat for your feedback. I have applied your suggestion.

jeana-redhat commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,85 @@
+$ oc -n openshift-sriov-network-operator get sriovnetworknodestate.sriovnetwork.openshift.io worker-0 -oyaml
+----
++
+[source,yaml]
```
Is this sample output?
```suggestion
.Example output
[source,yaml]
```

ogradyp replied:
Thanks @jeana-redhat for your feedback. Yes. I have applied your suggestion.

jeana-redhat commented on modules/nw-sriov-nic-mlx-secure-boot.adoc:
```diff
@@ -0,0 +1,85 @@
+----
+<1> The `totalvfs` value is the same number used in the `mstconfig` command earlier in the procedure. 
+
+. Enable secure boot to improve your system's security by preventing unauthorized operating systems and malicious software from loading during the device's boot process. 
```
Not sure if we want to make this claim
```suggestion
. Enable secure boot to prevent unauthorized operating systems and malicious software from loading during the device's boot process. 
```

ogradyp replied:
Thanks @jeana-redhat for your feedback. I have applied your suggestion.

jeana-redhat reviewed (commented):
a few nits here but overall seems good

/remove-label merge-review-in-progress
/remove-label merge-review-needed

ogradyp commented:
/label merge-review-needed

> a few nits here but overall seems good
> 
> /remove-label merge-review-in-progress /remove-label merge-review-needed

Thanks @jeana-redhat  for your feedback. I have squashed commits and addressed all your comments.

ogradyp commented:
/label merge-review-needed

sheriff-rh reviewed (approved):
LGTM; merging to 4.18+.

sheriff-rh commented:
/cherrypick enterprise-4.18

ogradyp commented:
Thanks @sheriff-rh for merging the content so quickly. Much appreciated.
